### PR TITLE
KiCad: 5.1.12 -> 6.0.0; and a bunch of cleanup

### PR DESCRIPTION
**Pull request by evils**
###### Motivation for this change
the first new mayor version release in 3 years!

###### Things done
* update the update script to handle a stray `6.9.9` tag that was created instead of `6.99` (unstable)
* update stable from `5.1.12` to `6.0.0`
  * removed the 5.x specific build conditions
  * remove the `withOCE` option
  * add the `withPCM` option (Plugins and Content Manager) (on by default)
  * add the `withI18n` option to `kicad.base` to build i18n there (off by default)
  * use `wxGTK31-gtk3` and `wxPython_4_1`, just like `kicad-unstable`
* update unstable from `6.0.0-rc1` to `2021-12-23` (latest rev that passes tests (at the time of writing))
  * with whatever libraries are the latest on `2021-12-25`
* remove the separate `i18n` stuff from the package, update script, etc
  * this was integrated into the source for unstable a while ago, and is now also in stable
  * does this need a warning about the removed `i18n` and `withI18n` inputs to `kicad.base`?
* had some more fun with `shellcheck`, added a shebang, cleared all notes
* correct build flag handling for `withNgspice`
* clean up some old package options and removed the associated asserts

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by prusnak** [comment]
Shouldn't we just use `wxGTK = wxGTK31-gtk3` and `wxPython = wxPython_4_1` with no `if (stable) ... else ...` ?

**Comment by evils** [comment]
> Shouldn't we just use `wxGTK = wxGTK31-gtk3` and `wxPython = wxPython_4_1` with no `if (stable) ... else ...` ?

"just", there's at least 9 attributes providing `wxWidgets`...
and i've had compatibility issues with mixing them with versions of `wxPython`

i'm going to look into if the reasons for the specific versions still apply
(i think it at least used to be the wx combo for unstable didn't support the python terminal but brought high DPI)

\</rant>

thanks for pointing that out, i forgot the reason for it may not apply anymore

**Comment by evils** [comment]
i did a bunch more cleanup, most of it could have probably be separated from the update, but a lot of it ended up in the 6.0.0 update commit

File: pkgs/applications/science/electronics/kicad/base.nix [hunk 1]
```diff
@@ -82,7 +82,7 @@ stdenv.mkDerivation rec {
   ++ optionals (!withScripting) [
     "-DKICAD_SCRIPTING_WXPYTHON=OFF"
   ]
-  ++ optional (withNgspice) "-DKICAD_SPICE=ON"
+  ++ optional (!withNgspice) "-DKICAD_SPICE=OFF"
```
Comment (veprbl) [inline pkgs/applications/science/electronics/kicad/base.nix:85]: An opportunity to just remove an option that never worked.

**Comment by veprbl** [comment]
@GrahamcOfBorg build kicad-small kicad-unstable-small python3Packages.kicad

**Comment by veprbl** [comment]
@GrahamcOfBorg build kicad-small python3Packages.kicad

File: pkgs/applications/science/electronics/kicad/base.nix (hunk 1 above)
Comment (evils) [reply pkgs/applications/science/electronics/kicad/base.nix:85]: why would i do that if i can make it work instead?

**Comment by evils** [comment]
the kicad-small on x86_64-linux failure is an intermittent test failure
i was hoping it'd be fixed before the release ([i've got an issue open about it](https://gitlab.com/kicad/code/kicad/-/issues/9888))
maybe i should just disable the tests...

**Comment by evils** [comment]
tests disabled (mostly because the current nixpkgs channel doesn't provide kicad due to that failure)
despite newer kicad master branch commits build, i'm sticking to the latest kicad-unstable which passes all those tests

**Comment by veprbl** [comment]
@GrahamcOfBorg build kicad-small kicad-unstable-small python3Packages.kicad

File: pkgs/applications/science/electronics/kicad/base.nix (hunk 1 above)
Comment (veprbl) [reply pkgs/applications/science/electronics/kicad/base.nix:85]: There is no point to keep so many knobs that are not tested. I think we can give our users some credit, to say that they can figure out how to fine tune kicad with `overrideAttrs` if they need to.

**Review by newAM — APPROVED** [review]
Used this all day yesterday on x86_64 NixOS.  Schematic capture, board layout, and 3D viewer all work great.

**Pull request merged**